tools/Hooks/pre-commit.sh

```diff
@@ -35,6 +35,26 @@ printf '%s\0' "${commit_files[@]}" | run_checks "${standard_checks[@]}"
 # Use git-clang-format to check for any suspicious formatting of code.
 @PYTHON_EXECUTABLE@ @CMAKE_SOURCE_DIR@/.git/hooks/ClangFormat.py
 
+###############################################################################
+# Use yapf to check python file formatting, only if it is installed.
+if command -v @YAPF_EXECUTABLE@ >/dev/null 2>&1; then
+    python_files=()
+
+    for commit_file in ${commit_files[@]}
+    do
+        echo $commit_file
+        if [[ $commit_file =~ .*\.py$ ]]; then
+            python_files+=("${commit_file}")
+        fi
+    done
+
+    if [ ${#python_files[@]} -ne 0 ]; then
+        @YAPF_EXECUTABLE@ -d ${commit_files[@]}
```

Comment:
Should this be `python_files` rather than `commit_files`?  I don't know if it matters.